gnocchiclient/exceptions.py

```diff
@@ -251,7 +251,7 @@ def from_response(response, method=None):
                 desc = body.get('description')
                 if desc and isinstance(desc, six.text_type):
                     for enhanced_cls in enhanced_classes:
-                        if enhanced_cls.match.match(desc):
+                        if enhanced_cls.match.match(str(desc)):
```

Comment:
not sure what this does since L252 checks that desc is a string
test?

Reply:
It is not actually a string, it is a six.test_type object. This causes a problem when using with the `match` operation that expects an actual string object.

Reply:
```
Python 3.9.1 (default, Dec 24 2020, 16:53:18)
[Clang 12.0.0 (clang-1200.0.32.28)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
>>> six.text_type
<class 'str'>
```

😕

Reply:
The problem happens with Python 2.7. 

```
**~>  python
Python 2.7.17 (default, Dec 29 2020, 07:44:13) 
[GCC 9.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
>>> six.text_type
<type 'unicode'>
**
```

Reply:
We don't support Python 2 anymore.

Reply:
I see. Is Gnocchi announcing that somewhere? Otherwise people will not be aware of that.

Reply:
It's in `setup.cfg` 😉